src/test/java/io/phasetwo/keycloak/events/WebhookSenderEventListenerProviderTest.java

```diff
@@ -0,0 +1,281 @@
+    Response clientCreationResponse = realm.clients().create(client);
+    assertThat(clientCreationResponse.getStatus(), is(201));
+
+    // Adding rights
+    String realmManagementId = realm.clients().findByClientId("realm-management").get(0).getId();
+    String clientId = realm.clients().findByClientId(realmAdminClientId).get(0).getId();
+    String serviceUserId = realm.clients().get(clientId).getServiceAccountUser().getId();
+    List<RoleRepresentation> availableRoles = realm
+            .users()
+            .get(serviceUserId)
+            .roles()
+            .clientLevel(realmManagementId)
+            .listAvailable();
+    List<RoleRepresentation> rolesToAssign = availableRoles
+            .stream()
+            .filter(r ->
+                "realm-admin".equalsIgnoreCase(r.getName()))
+            .collect(Collectors.toList());
+    assertThat(rolesToAssign.size(), is(1));
+    realm.users().get(serviceUserId).roles().clientLevel(realmManagementId).add(rolesToAssign);
+
+    AtomicReference<String> body = new AtomicReference<String>();
+    // create a server on a free port with a handler to listen for the event
+    int port = WEBHOOK_SERVER_PORT;
+    createWebhook(
+        keycloak,
+        httpClient,
+        webhookUrl(TEST_REALM),
+        "http://host.testcontainers.internal:" + port + "/webhook",
+        "qlfwemke",
+        ImmutableSet.of("admin.*"));
+
+    Server server = new Server(port);
+    server
+        .router()
+        .POST(
+            "/webhook",
+            (request, response) -> {
+              String r = request.body();
+              log.infof("%s", r);
+              body.set(r);
+              response.body("OK");
+              response.status(202);
+            });
+    server.start();
+    Thread.sleep(1000l);
+
+    try {
+        // log in to the test realm
+        Keycloak keycloakTestRealm = KeycloakBuilder.builder()
+                .serverUrl(getAuthUrl())
+                .clientId(realmAdminClientId)
+                .clientSecret(realmAdminClientSecret)
+                .realm(TEST_REALM)
+                .grantType(OAuth2Constants.CLIENT_CREDENTIALS)
+                .build();
+        realm = keycloakTestRealm.realm(TEST_REALM);
+
+	    // cause an event to be sent
+	    UserRepresentation userRepresentation = new UserRepresentation();
+	    userRepresentation.setUsername("username");
+	    Response userResponse = realm.users().create(userRepresentation);
+	    assertThat(userResponse.getStatus(), is(201));
+
+	    Thread.sleep(1000l);
+
+	    // check the handler for the event, after a delay
+	    String receivedPayload = body.get();
+	    ExtendedAdminEvent event = parseEvent(receivedPayload);
+	    assertThat(event.getRealmId(), equalTo(TEST_REALM));
+	    assertThat(event.getAuthDetails().getRealmId(), equalTo(TEST_REALM));
+
+	    // cleanup
+	    realm = keycloak.realm(TEST_REALM);
+	    realm.remove();
+    }
+    finally {
+        server.stop();
+    }
+  }
+
```

Comment:
We could add a new test like : "a **master** user realm creating a user on a **test-realm** and then a user from **test-realm** is deleting it. This should cover all possible scenarios

Reply:
Sure, added.